torch/_decomp/decompositions.py

```diff
@@ -286,6 +286,22 @@ def prelu_backward(
     return (input_grad.view_as(self), out)
 
 
+@register_decomposition(aten._prelu_kernel)
+@pw_cast_for_opmath
```

Comment:
pw_cast_for_opmath not needed here, it only leads to more complicated IR

Reply:
We are outputting either `self` or `weight * self`, in both cases there's no difference whether they are pre-converted to float or not here - `self` is a copy either way, `weight * self` would do whatever is needed in multiplication

Reply:
I'll clean up all that code then.